genie-web/src/main/java/com/netflix/genie/web/scripts/ResourceSelectorScript.java

```diff
@@ -41,8 +65,50 @@
      * @return A {@link ResourceSelectorScriptResult} instance
      * @throws ResourceSelectionException If an unexpected error occurs during selection
      */
-    ResourceSelectorScriptResult<R> selectResource(
+    public ResourceSelectorScriptResult<R> selectResource(
+        final Set<R> resources,
+        final JobRequest jobRequest
+    ) throws ResourceSelectionException {
+        try {
+            final Map<String, Object> parameters = Maps.newHashMap();
+            parameters.put(JOB_REQUEST_BINDING, jobRequest);
+            this.addParametersForScript(parameters, resources, jobRequest);
+            final Object evaluationResult = this.evaluateScript(parameters);
+            if (!(evaluationResult instanceof ResourceSelectorScriptResult)) {
+                throw new ResourceSelectionException(
+                    "Selector evaluation returned invalid type: " + evaluationResult.getClass().getName()
+                        + " expected " + ResourceSelectorScriptResult.class.getName()
+                );
+            }
+            @SuppressWarnings("unchecked") final ResourceSelectorScriptResult<R> result
+                = (ResourceSelectorScriptResult<R>) evaluationResult;
+
+            // Validate that the selected resource is actually in the original set
+            if (result.getResource().isPresent() && !resources.contains(result.getResource().get())) {
+                throw new ResourceSelectionException(result.getResource().get() + " is not in original set");
+            }
+
+            return result;
+        } catch (
+            final ScriptExecutionException
+                | ScriptNotConfiguredException
+                | RuntimeException e
+        ) {
+            throw new ResourceSelectionException(e);
+        }
+    }
+
+    /**
+     * Add any implementation specific parameters to the map of parameters to send to the script. The job request
+     * will already have been added under {@literal jobRequestParameter}.
+     *
+     * @param parameters The existing set of parameters for implementations to add to
+     * @param resources  The set of resources that need to be chosen from
+     * @param jobRequest The job request necessitating the evaluation of this script
+     */
+    protected abstract void addParametersForScript(
+        Map<String, Object> parameters,
         Set<R> resources,
         JobRequest jobRequest
-    ) throws ResourceSelectionException;
+    );
```

Comment:
It's covered by the extensions for clusters and commands. Since they were so extensive I didn't feel a need to copy and paste them over for more redundancy